### DISCUSSION TITLE
About Licenses

### DISCUSSION THREAD

**Discussion by ndrean** [dwyl/image-classifier#111] [General]
I see that `Bumblebee` uses APACHE whilst you use GNU.

I read that:
- MIT: "do what you want with this, just don’t sue me". This means you can even derive a closed commercial product from this.
- APACHE: "do what you want with this, just don’t sue me. You can derive anything but keep the same license and mark the changes."
- GNU: "do what you want with this,don’t sue me and keep it open". If you derive something from this, it has to be open source."

Do you agree on these definitions?

**Comment by nelsonic** [comment]
`GPL` doesn't require derivative works to be open source; it's encouraged.
Plenty of commercial products use the [Linux Kernel](https://en.wikipedia.org/wiki/Linux) and are _not_ Open Source.
See: https://opensource.stackexchange.com/questions/2338/can-i-use-gpl-libraries-in-a-closed-source-project

It's true that `MIT` and `Apache` are _less_ restrictive but only from a narrow-minded copyright lawyer's perspective.
`Google`, `Cisco`, `Oracle` and even `Microsoft` use `GPL` code in their products. Don't fear `GPL`.

**Comment by ndrean** [comment]
Thanks for your input And GREAT link. Will take me time to digest it.
